src/games/strategy/triplea/delegate/BattleDelegate.java

```diff
@@ -100,6 +100,75 @@ public void start() {
       addBombardmentSources();
       m_needToAddBombardmentSources = false;
     }
+    
+    Territory lastAmphib = null, lastNonAmphib = null, t;
+    IBattle battle;
+    Iterator<Territory> battleTerritories = m_battleTracker.getPendingBattleSites(true).iterator(); // get bombing/air raids
+    if( battleTerritories.hasNext() ) {
+      t = battleTerritories.next();
+      if( battleTerritories.hasNext() ) return; // If more than one air/boming raid do nothing here
+      battle = m_battleTracker.getPendingBattle(t, true, null);
+      if( battle == null ) {
+        System.out.println("Air/Bombing Raid gone missing in BattleDelegate");
```

Comment:
`Preconditions.checkNotNull` is a good library to check that. Best yet is to ensure that the value actually will never be null in which case you do not need the check.

Iterators are for looping over a set of values. You are simply using it to peek at the first value, can be done with a size check instead. A Foreach loop is then not even terribly appropriate (grant it, you could easily break from that and not need to count)

Normal java code is a bit immaterial. Please use the code formatter, and follow the conventions that were agreed upon by the development group.  Last, consistent with any OO language, there are ways to have good quality and others where the code is procedural and may as well be C. In the latter case, after merge, it is the entire development group that is responsible for the code, not just one person.  Not saying that this is impossible to maintain, but there are a number of procedural elements that does not make this the easiest.